### PR TITLE
Feature: history

### LINKED ISSUE #64: Add history feature

### PR DESCRIPTION
**Pull request by jamesjwarren**
# Feature: History

This PR includes:
- `FM.history` module
- `/history` view - closes #64 
- infinite scrolling of history list

### CODE REVIEW

File: app/js/history/controllers/HistoryCtrl.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                resolve: {
+                    historyData: ["PlayerHistoryResource", "$q", "$route", function (PlayerHistoryResource, $q, $route){
+                        var deferred = $q.defer();
+
+                        // resolve headers from query
+                        PlayerHistoryResource.query($route.current.params, function(response, headers){
+                            deferred.resolve({
+                                data: response,
+                                headers: headers
+                            });
```
Comment (jamesjwarren) [inline app/js/history/controllers/HistoryCtrl.js]: Seemed like the neatest way to access response headers which we need for pagination headers.

Comment (eopa) [inline app/js/history/controllers/HistoryCtrl.js]: you should avoid putting logic like this in the config, if you need to serialize the data it should be done in the resource using transformResponse function, like with the `playing` flag from the transport resource

File: app/index.html [hunk 2]
```diff
@@ -53,6 +53,7 @@
 
         <!-- build:js js/app.min.js -->
         <script src="components/socket.io-client/socket.io.js"></script>
+        <script src="components/jquery/dist/jquery.js"></script>
```
Comment (eopa) [inline app/index.html]: Do we really need to use jquery for this :cry: ? I know `ngInfiniteScroll` depends on this but it seems a bit over the top to have to include this library just for this module. Maybe we should explore alternatives or remove infinite scrolling and use regular pagination instead.

Comment (jamesjwarren) [inline app/index.html]: Yeh I agree, I really didn't want to have to include it either. I'll look into alternative solutions.

**Comment by jamesjwarren** [comment]
I've replaced the previous implementation with a simple directive `snInfiniteScroll` that detects if an element is scrolled to the end. Just need to write some tests for it.

File: app/js/infinite-scroll/infiniteScroll.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                    if (elementScrollEnd && !$scope.disabled) {
+                        $scope.snInfiniteScroll();
+                    }
+
+                };
+
+                var removeListeners = function removeListeners(){
+                    angular.element(scrollingElement).off("scroll");
+                    angular.element($window).off("resize");
+                };
```
Comment (eopa) [inline app/js/infinite-scroll/infiniteScroll.js]: you should document your code to explain what it does

**Comment by jamesjwarren** [comment]
@eopa docs and tests for snInfiniteScroll have been updated

**Comment by eopa** [comment]
:+1: LGTM

**Pull request merged**